lib/utils.c

```diff
@@ -39,6 +39,54 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[41], buf[80], *endp;
+	ssize_t len;
+	FILE *fp;
+	long result;
+
+	sprintf(fname, "/sys/class/net/%s/%s", dev, prop);
+	fp = fopen(fname, "r");
+	if (fp == NULL) {
+		if (strcasecmp(prop, "tun_flags"))
+			fprintf(stderr, "open %s: %s\n", fname,
+				strerror(errno));
+		return -1;
+	}
+
+	len = fread(buf, sizeof(buf), 1, fp);
+	fclose(fp);
+	if (len < 0) {
+		fprintf(stderr, "read %s: %s", fname, strerror(errno));
+		return -1;
+	}
+
+	result = strtoul(buf, &endp, 0);
```

Comment:
I'll need to look more into the manual to get the return value here right, but in your next rebase if you have time you should try it as well. Looking at examples of uses of `strtoul()` in this repository or the kernel may help.